### PR TITLE
WIP: clang-format for code style checking/enforcing

### PR DESCRIPTION
**Pull request by ayyaruq**
[Clang Format](https://clang.llvm.org/docs/ClangFormat.html) is a tool to automatically reformat code against a fixed style configuration. With feedback in Discord, I've got the config to reflect our style as closely as I can. Some changes are really minor and mostly around whitespacing, others (like tools and the older parts of the zone server or current wip implementations) are a bit larger.

This PR is still a work in progress. Each server plus the tools are in a separate commit, so when reviewing please look at individual commits rather than the full PR. Alternatively it can be split into a separate PR to bring each server into consistency. Leave feedback here and we'll figure something out.

### CODE REVIEW

File: src/servers/Server_Zone/Actor/PlayerQuest.cpp [hunk 1]
```diff
@@ -408,7 +406,7 @@ uint16_t Core::Entity::Player::getQuestUI16A( uint16_t questId )
    if( idx != -1 )
    {
       boost::shared_ptr< QuestActive > pNewQuest = m_activeQuests[idx];
```
Comment (GokuWeedLord) [inline src/servers/Server_Zone/Actor/PlayerQuest.cpp]: spacing inside square brackets?

File: src/servers/Server_Zone/Network/Handlers/GMCommandHandlers.cpp [hunk 2]
```diff
@@ -184,30 +182,27 @@ void Core::Network::GameConnection::gm1Handler( const Packets::GamePacket& inPac
...
-         "\nExp: " + std::to_string( targetPlayer->getExp() ) +
-         "\nSearchMessage: " + targetPlayer->getSearchMessage() +
-         "\nPlayTime: " + std::to_string( targetPlayer->getPlayTime() ) );
+      player.sendNotice(
+          "Name: " + targetPlayer->getName() + "\nGil: " + std::to_string( targetPlayer->getCurrency( 1 ) ) +
+          "\nZone: " + targetPlayer->getCurrentZone()->getName() + "(" + std::to_string( targetPlayer->getZoneId() ) +
+          ")" + "\nClass: " + std::to_string( static_cast< uint8_t >( targetPlayer->getClass() ) ) + "\nLevel: " +
+          std::to_string( targetPlayer->getLevel() ) + "\nExp: " + std::to_string( targetPlayer->getExp() ) +
+          "\nSearchMessage: " + targetPlayer->getSearchMessage() +
+          "\nPlayTime: " + std::to_string( targetPlayer->getPlayTime() ) );
```
Comment (GokuWeedLord) [inline src/servers/Server_Zone/Network/Handlers/GMCommandHandlers.cpp]: this is basically worse than what we had but there's probably nothing that can be done without rewriting this or disabling formatting for this one part right?

Comment (SapphireMordred) [reply src/servers/Server_Zone/Network/Handlers/GMCommandHandlers.cpp]: Yea, it wont be able to properly format that automatically.

**Comment by SapphireMordred** [comment]
Judging by the CI output, did it remove includes?

**Comment by SapphireMordred** [comment]
Not applicable anymore.

**Pull request closed without merging**